fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java

```diff
@@ -0,0 +1,241 @@
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.support.TransactionCallback;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.sql.DataSource;
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
+
+
+/**
+ * An implementation of the {@link SearchIndex}
+ *
+ * @author dbernstein
+ */
+@Component
+public class DbSearchIndexImpl implements SearchIndex {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DbSearchIndexImpl.class);
+    private static final String SIMPLE_SEARCH_TABLE = "simple_search";
+    private static final String DDL = "sql/default-search-index.sql";
+
+    @Inject
+    private DataSource dataSource;
+
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Inject
+    private PlatformTransactionManager platformTransactionManager;
+
+    /**
+     * Setup database table and connection
+     */
+    @PostConstruct
+    public void setup() {
+        LOGGER.info("Applying ddl: {}", DDL);
+        DatabasePopulatorUtils.execute(
+                new ResourceDatabasePopulator(new DefaultResourceLoader().getResource("classpath:" + DDL)),
+                this.dataSource);
+        this.jdbcTemplate = getNamedParameterJdbcTemplate();
+    }
+
+    private NamedParameterJdbcTemplate getNamedParameterJdbcTemplate() {
+        return new NamedParameterJdbcTemplate(this.dataSource);
+    }
+
+    @Override
+    public SearchResult doSearch(final SearchParameters parameters) throws InvalidQueryException {
+        //translate parameters into a SQL query
+        var paramCount = 1;
+
+        final MapSqlParameterSource parameterSource = new MapSqlParameterSource();
+        final var whereClauses = new ArrayList<String>();
+        for (Condition condition : parameters.getConditions()) {
+            final var field = condition.getField();
+            final var operation = condition.getOperator();
+            var object = condition.getObject();
+            if (field.equals(Condition.Field.FEDORA_ID) &&
+                    condition.getOperator().equals(Condition.Operator.EQ)) {
+                if (!object.equals("*")) {
+                    if (object.contains("*")) {
+                        object = object.replace("*", "%");
+                        whereClauses.add(Condition.Field.FEDORA_ID + " like '" + object + "'");
```

Comment:
Do we need to take any sql injection precautions here?

Reply:
Yes - good catch - I'll use parameterSource for those.